src/syntax.c

```diff
@@ -1505,7 +1507,7 @@ static struct value *compile_concat(struct term *exp, struct ctx *ctx) {
 
     struct type *t = exp->type;
     struct info *info = exp->info;
-    struct value *v;
+    struct value *v = NULL;
```

Comment:
v is assigned in every branch of the subsequent if-else-if... structure, so this change in conjunction with removing line 1549 (below) is unneccessary

I would leave this unchanged. because it's more a question of style than making an improvement.